### PR TITLE
Refactor release tooling code into subfolders

### PR DESCRIPTION
**Pull request by abhay-krishna**
The release tooling code has some considerable techdebt in the form of duplicated code, misplaced functions, monolithic `pkg` directory with a variety of functionalities. This PR (WIP) aims to address a portion of it: separating the pkg directory into functionality-based subfolders and finding a proper home for the functions.

/hold
/milestone techdebt

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: config/crd/bases/anywhere.eks.amazonaws.com_tinkerbelldatacenterconfigs.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+
+---
+apiVersion: apiextensions.k8s.io/v1
```
Comment (g-gaston) [inline config/crd/bases/anywhere.eks.amazonaws.com_tinkerbelldatacenterconfigs.yaml:3]: sorry to be a pain but can we move these two to a separate PR (I don't mind taking it)?

File: release/pkg/image/image.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
+package image
```
Comment (g-gaston) [inline release/pkg/image/image.go:1]: nit: `images`?

File: config/crd/bases/anywhere.eks.amazonaws.com_tinkerbelldatacenterconfigs.yaml (hunk 1 above)
Comment (abhay-krishna) [reply config/crd/bases/anywhere.eks.amazonaws.com_tinkerbelldatacenterconfigs.yaml]: Sureee
